lib/ansible/modules/system/known_hosts.py

```diff
@@ -145,7 +148,7 @@ def enforce_state(module, params):
                 module.fail_json(msg="Failed to read %s: %s" % \
                                      (path,str(e)))
         try:
-            outf=tempfile.NamedTemporaryFile(dir=os.path.dirname(path))
+            outf = tempfile.NamedTemporaryFile(mode='w+', dir=os.path.dirname(path))
```

Comment:
The reason those changes are there:
- I was asked to write an integration test
- the integration test uncovered some minor buglets in the code
- I had to fix those buglets to make the new integration test pass

Incidentally, the extra buglets were these two:
- the known_hosts module didn't work under Python 3 (not a blocker, since modules aren't expected to work under Python 3, but I didn't like the yellow blob in Shippable and the fix wasn't large)
- the known_hosts module failed with an error if you specified `hash_host=true` and had a relatively recent OpenSSH version (6.4 or newer) -- looks like this was already filed as bug #20500.

Would you like me to extract the integration test and the bugfixes into a separate PR?  I would prefer not to (it's quite a bit of extra work for me), but I can be easily convinced with a little bit of blackmail (e.g. "we won't merge this unless you do that").